packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -106,10 +106,10 @@ export let visitor = {
     argsId._shadowedFunctionLiteral = path;
 
     function optimiseCandidate(parent, parentPath, offset) {
-      if (t.isReturnStatement(parentPath.parent) || t.isIdentifier(parentPath.parent.id)) {
+      if (t.isReturnStatement(parentPath.parent) || t.isIdentifier(parentPath.parent.id) || t.isIdentifier(parentPath.parent.left)) {
```

Comment:
Thanks a lot for the review, I really appreciate it!

Here, `left` was referring to the left-hand side of an assignment operation. It was way too complicated. Covering all cases in this condition is an impossible task.

I don't understand your second comment about `parent` and `parentPath`. I need `parent` to get to the properties and `parentPath` to `replaceWith`. Is there a workaround?